buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixtureExtension.java

```diff
@@ -42,6 +85,20 @@ public void useFixture(String path) {
             );
         }
         fixtures.add(fixtureProject);
+        // Check for exclusive access
+        if (globalMap.containsKey(path)) {
+            throw new GradleException("Projects " + globalMap.get(path) + " and " + this.project.getPath() + " both " +
+                "claim all services from " + path + ". This is not supported because it breaks running in parallel. " +
+                "Configure specific services in docker-compose.yml for each and add the service name to `useFixture`"
+            );
+
+        }
     }
 
+    boolean isServiceInUse(String serviceName, String fixtureProject) {
```

Comment:
The name `serviceInUse` implies this is some kind of runtime check. Probably something like `isServiceRequired` would make more sense.